src/forms/NumberInput.jsx

```diff
@@ -31,73 +69,77 @@ export class NumberInput extends React.Component {
 		this.onKeyDown = this.onKeyDown.bind(this);
 	}
 
-	static getDerivedStateFromProps(nextProps, prevState) {
-		const isNewValue =
-			nextProps.onChange && nextProps.value !== prevState.value;
+	static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+		const isNewValue = nextProps.onChange && nextProps.value !== prevState.value;
 
 		return {
 			value: isNewValue ? nextProps.value : prevState.value,
 		};
 	}
 
-	_updateValueByStep(isIncreasing) {
-		const currentVal = new Number(this.state.value);
-		const step = new Number(this.props.step);
+	_updateValueByStep: Function;
+	_updateValueByStep(isIncreasing: boolean) {
+		const currentVal = parseInt(this.state.value);
+		const step = parseInt(this.props.step);
 		let newValue = isIncreasing ? currentVal + step : currentVal - step;
 
-		if (newValue > this.props.max) {
+		if (newValue < this.props.min) {
+			return this.props.min;
+		}
+
+		if (this.props.max && newValue > this.props.max) {
 			newValue = this.props.max;
-		} else if (newValue < this.props.min) {
-			newValue = this.props.min;
 		}
 
 		return newValue;
```

Comment:
these `if` statements are just a workaround by initial author who didn't know how to write a min-max expression - let's use this:

```js
const newValue = isIncreasing ? currentVal + step : currentVal - step;
return Math.min(Math.max(newValue, this.props.min), this.props.max || Infinity);
```

feel free to make an intermediate variable to make it a bit more legible

```js
const newValue = isIncreasing ? currentVal + step : currentVal - step;
const minConstrained = Math.max(newValue, this.props.min);
return Math.min(minConstrained, this.props.max || Infinity);
```